### ISSUE TITLE
Feature Request: Subnet Grouping for Connections View

### ISSUE THREAD

**Issue by mmguero** [aol/moloch#881]
For large networks with many hosts, the connections view can get cluttered. It would be useful to be able to specify a CIDR mask, something like "Src IP/24", "Dst IP/16", etc. for the "Field for the source/destination nodes" in the connections view. Then, the connections view could display which subnets are talking to each other, rather than just individual hosts.

I think you could possibly use Elasticsearch's [IP Range Aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-iprange-aggregation.html) for this.

There would have to be some thought about how IPv6 would fit into it. I guess you could just scale the subnet mask up from 32 to 128 bits (eg., if they did Src IP/16 that would be equivalent to /64 for IPv6 addresses?).

**Comment by mmguero** [comment]
I think if using IP Range Aggregation isn't going to pan out, you could just do the query like you normally would and apply the subnet range manually to the IPs prior to putting them in their buckets in updateValues in buildConnections in viewer.js.

This feature would give you kind of a "network zones overview" in the connections view similar to what GrassMarlin has in the Logical Graph. It would be particularly useful as a quick overview to see if there is cross-segment communication in an environment where there should not be (for example, is the control system/OT environment communicating with the corporate/IT environment if it should not be).

**Comment by problematiq** [comment]
Generally most of us use WISE and an iPAM for this or we segregate the networks via capture nodes. I like the idea of Moloch having it built in, but I fear if you have overlapping network subnets on different vlans, it might become confusing.

**Comment by mmguero** [comment]
Yeah, I agree not every feature is useful for every user. I'm not saying a subnet-grouped connections view would be something that should be the default selection, but similar to how it's possible to use various combinations of source and destination fields, I think this would be a useful option.

**Comment by WilliamSalusky** [comment]
👍
